components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java

```diff
@@ -40,6 +42,25 @@ public String retrieveArtifact(String APIId, String gatewayLabel, String gateway
         return gatewayRuntimeArtifacts;
     }
 
+    @Override
+    public List<String> retrieveAllArtifacts(String label) throws ArtifactSynchronizerException {
+        List<String> gatewayRuntimeArtifactsArray = new ArrayList<>();
+        try {
+            List<ByteArrayInputStream> baip = apiMgtDAO.getAllGatewayPublishedAPIArtifacts(label);
+            for (ByteArrayInputStream byteStream :baip){
+                byte[] bytes = ByteStreams.toByteArray(byteStream);
+                String  gatewayRuntimeArtifacts = new String(bytes);
+                gatewayRuntimeArtifactsArray.add(gatewayRuntimeArtifacts);
+            }
+            if (log.isDebugEnabled()){
```

Comment:
No need of using log.isDebugEnabled() if no string concatenation takes place

Reply:
Removed in the latest version